language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/constants/PatternConstants.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ballerinalang.langserver.common.constants;
+
+/**
+ * Holds constants related to regex patterns.
+ *
+ * @since 2.0.0
+ */
+public class PatternConstants {
```

Comment:
Any reason to duplicate the identifier matching logic here. Can't we reuse something that Parser/Runtime uses for this? My point is any changes to identifier matching patterns will not reflected here and we might need to maintain this.

Reply:
I had the same doubt. Do you know where the identifier matching logic is in the parser?

Reply:
there's a similar class called [IdentifierUtils](https://github.com/ballerina-platform/ballerina-lang/blob/master/bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/utils/IdentifierUtils.java), but need to verify it from the compiler/runtime team. @lochana-chathura @rdulmina anyway to derive `isIdentifier()` capability through Compiler-APIs?

Reply:
We do the identifier matching here in the lexer. https://github.com/ballerina-platform/ballerina-lang/blob/317188576353a17d5abbde7fb1e3bb153811d0ac/compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaLexer.java#L783

It is not merely an `isIdentifier(String txt)` method. We read the input sequence char by char using the attached token reader to capture the identifier. So cannot expose as it is. But, maybe we can write a method to boot up a parser and try to parse an identifier from a given string and then return whether it is valid. But, currently, we don't have a place to expose such API.

Reply:
In order to have a common approach we need to expose it via the Syntax API. Hence I'm +1 for the current approach which @IMS94 has followed.